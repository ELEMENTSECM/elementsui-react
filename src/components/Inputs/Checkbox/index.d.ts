import * as React from 'react';

export interface CheckboxProps {
        /**
         * Checkbox label
         */
        label?: string;
        /**
         * Checkbox is disabled
         */
        disabled?: boolean;
        /**
         * Checkbox description
         */
        ariaDescribedBy?: string;
    }

    export const Checkbox: React.SFC<CheckboxProps>;


