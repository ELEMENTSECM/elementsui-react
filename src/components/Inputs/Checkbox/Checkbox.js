import * as React from 'react';
import PropTypes from 'prop-types';
import { Checkbox as UIFabCheckbox } from 'office-ui-fabric-react/lib/Checkbox';
import { styles } from './Checkbox.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** Checkbox example */
export function Checkbox(props) {
	const { htmlId, label, disabled, styles } = props;
	const classNames = classNamesFunction()(styles, props);
	return (
		<UIFabCheckbox
			id={htmlId}
			className={classNames.root}
			label={label}
			disabled={disabled}
			ariaDescribedBy={'descriptionID'}
		/>
	);
}

Checkbox.propTypes = {
	/** HTML id tag of the root element */
	htmlId: PropTypes.string,
	/** Checkbox label */
	label: PropTypes.string,
	/** Checkbox is disabled */
	disabled: PropTypes.bool,
	/** Checkbox description */
	ariaDescribedBy: PropTypes.string,
	/** User-defined styling */
	styles: PropTypes.func
};

export default styled(customizable('Checkbox', ['theme'])(Checkbox), styles);
