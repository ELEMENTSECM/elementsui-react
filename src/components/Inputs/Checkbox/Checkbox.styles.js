export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Checkbox',
			{
				// place your styles here
			},
			className
		]
	};
};
