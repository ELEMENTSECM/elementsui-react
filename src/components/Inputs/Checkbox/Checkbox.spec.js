import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
	test('displays `Foo` as label', () => {
		const tree = renderer
			.create(<Checkbox id="defaultCheckbox" label="" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
