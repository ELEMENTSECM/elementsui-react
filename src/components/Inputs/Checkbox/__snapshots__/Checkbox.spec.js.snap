// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox displays \`Foo\` as label 1`] = `
<button
  aria-checked={false}
  aria-describedby="descriptionID"
  aria-disabled={undefined}
  aria-label={undefined}
  aria-labelledby={undefined}
  aria-posinset={undefined}
  aria-setsize={undefined}
  className="ms-Checkbox is-enabled css-32"
  disabled={undefined}
  id="checkbox-0"
  name={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  role="checkbox"
  type="button"
>
  <label
    className="ms-Checkbox-label css-33"
    htmlFor="checkbox-0"
  >
    <div
      className="ms-Checkbox-checkbox css-34"
    >
      <i
        aria-hidden={true}
        className="root-37 css-35"
        data-icon-name="CheckMark"
        role="presentation"
      />
    </div>
  </label>
</button>
`;
